### PR TITLE
METAL-745: Make CBO aware of HostFirmwareComponents

### PR DESCRIPTION
**Pull request by iurygregory**
This PR updates BMO to point to the latest version we have downstream,
the version includes changes that are necessary for
HostFirmwareComponents.

⚠️  **NOTE: with controller-runtime 0.16.3  a lot of things changed on how
we can specify NewCache, I'm attempting to simplify things by using
`Cache` instead of `NewCache`. 
I was necessary to update the FakeClient used in our testing**⚠️ 

Also includes hostfirmwarecomponents in relatedObjects for the
clusteroperator controller.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/openshift/cluster-baremetal-operator
 
-go 1.19
+go 1.21
+
+toolchain go1.21.6
```
Comment (dtantsur) [inline go.mod:5]: This is not necessary in reality, let's remove.

**Comment by iurygregory** [comment]
/payload-job periodic-ci-openshift-release-master-nightly-4.16-e2e-metal-ipi-ovn-serial-virtualmedia
/payload-job periodic-ci-openshift-release-master-nightly-4.16-e2e-metal-ovn-single-node-live-iso
/payload-job periodic-ci-openshift-release-master-nightly-4.16-e2e-metal-ipi-sdn-bm

**Comment by iurygregory** [comment]
proxyconnect tcp: dial tcp : i/o timeout
/test e2e-metal-ipi-serial-ipv4

**Comment by iurygregory** [comment]
- `periodic-ci-openshift-release-master-nightly-4.16-e2e-metal-ipi-ovn-serial-virtualmedia` Green ✅  on https://pr-payload-tests.ci.openshift.org/runs/ci/a1b500a0-0012-11ef-9910-fa6accfaa7c0-0

-  `periodic-ci-openshift-release-master-nightly-4.16-e2e-metal-ovn-single-node-live-iso` and `periodic-ci-openshift-release-master-nightly-4.16-e2e-metal-ipi-sdn-bm` Green ✅  on https://pr-payload-tests.ci.openshift.org/runs/ci/80e01420-ff66-11ee-9f42-befe5dd7d559-0

**Comment by iurygregory** [comment]
APIServerDeploymentDegraded: 1 of 3 requested instances are unavailable for apiserver.openshift-oauth-apiserver 

/test e2e-metal-ipi-upgrade

**Comment by iurygregory** [comment]
Couldn't really figure out the problem in the upgrade ipv6
retesting
/test e2e-metal-ipi-upgrade-ovn-ipv6

**Pull request closed without merging**